spec/components/candidate_interface/course_review_component_spec.rb

```diff
@@ -60,4 +60,41 @@
       end
     end
   end
+
+  describe 'degree requirements' do
+    let(:application_choice) { create(:application_choice, course_option: create(:course_option, course:)) }
+
+    context 'when course is undergraduate' do
+      let(:course) do
+        create(
+          :course,
+          :teacher_degree_apprenticeship,
+          degree_grade: 'two_one', # forcing a bad data coming from Publish
```

Comment:
Suggestion to be more descriptive with the 'context' and 'describe' rather than leaving comments in the codebase, thats what they are for after all.

Reply:
Makes sense. Done